Library/Homebrew/formula.rb

```diff
@@ -494,7 +494,16 @@ def python(options={}, &block)
 
   # an array of all {Formula} names
   def self.names
-    Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, '.rb' }.sort
+    core = Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, '.rb' }.sort
+    taps = Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"].reduce([]) do |formulae, tap|
+      Pathname.new(tap).find_formula do |f|
+        if f.to_s =~ HOMEBREW_TAP_PATH_REGEX
+           formulae << "#$1/#{$2.gsub(/^homebrew-/, "")}/#{f.basename(".rb")}"
+        end
+      end
+      formulae
+    end.sort
+    core+taps
   end
```

Comment:
I'd consider not returning the fully-qualified name here for taps and perhaps just add another method that does that. I think, despite it being a private API, that changing the output format for taps here will break some e.g. external commands.